### PR TITLE
Move Flatten Paged Results to Base Client

### PR DESCRIPTION
**Pull request by blueandgold**
There are some paged results which need 2 item_keys to extract the objects of interests, i.e. they are 2 layers deep.  These are not handled in this PR.

### CODE REVIEW

File: google/cloud/security/common/gcp_api/compute.py [hunk 1]
```diff
@@ -247,9 +187,11 @@ def get_instance_templates(self, project_id):
         instance_templates_api = self.service.instanceTemplates()
         list_request = instance_templates_api.list(
             project=project_id)
-        return self._flatten_list(
-            self._build_paged_result(list_request, instance_templates_api,
-                                     self.rate_limiter))
+
+        paged_resuls = self._build_paged_result(
```
Comment (carise) [inline google/cloud/security/common/gcp_api/compute.py]: should we name this "paged_results"?

File: google/cloud/security/common/gcp_api/compute.py [hunk 2]
```diff
@@ -247,9 +187,11 @@ def get_instance_templates(self, project_id):
...
+            list_request, instance_templates_api, self.rate_limiter)
+
+        return self._flatten_list_results(paged_resuls, 'items')
```
Comment (carise) [inline google/cloud/security/common/gcp_api/compute.py]: "paged_results"?

File: google/cloud/security/common/gcp_api/compute.py (hunk 1 above)
Comment (blueandgold) [inline google/cloud/security/common/gcp_api/compute.py]: Done

File: google/cloud/security/common/gcp_api/compute.py (hunk 2 above)
Comment (blueandgold) [inline google/cloud/security/common/gcp_api/compute.py]: Done

**Comment by blueandgold** [comment]
Adding `cla: yes` label, as the PR author is a googler at the time of this PR was created.

**Pull request merged**